File: 0169-majority-element/0169-majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        ele = None
        count = 0
        for num in nums:
            if ele == None or count == 0:
                ele = num
                count = 1
            
            elif ele == num:
                count+=1
            else:
                count-=1
            
            
            # if count == 0:
            #     ele = None
            # else:
            #     ele = num
        return ele
            

        

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int element = nums[0];
        int count = 1;
        for(int i=1;i<nums.length;i++){
            if(count==0){
                count+=1;
                element = nums[i];
            }
            else if(nums[i]==element) count+=1;
            else if(nums[i]!=element) count-=1;
        }
        return element;
    }
}